wire-schema/src/main/java/com/squareup/wire/schema/ProtoType.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema
+
+import com.google.common.base.Preconditions.checkNotNull
+import java.util.Collections
+import java.util.LinkedHashMap
+
+/**
+ * Names a protocol buffer message, enumerated type, service, map, or a scalar. This class models a
+ * fully-qualified name using the protocol buffer package.
+ */
+class ProtoType {
+  val isScalar: Boolean
+
+  private val string: String
+
+  val isMap: Boolean
+
+  /** The type of the map's keys. Only present when [.isMap] is true.  */
+  val keyType: ProtoType?
+
+  /** The type of the map's values. Only present when [.isMap] is true.  */
+  val valueType: ProtoType?
+
+  val simpleName: String
+    get() {
```

Comment:
Why not compute once and assign?

Reply:
To save allocations. This isn’t usually needed.